packages/lib/gpcircuits/circuits/ownerV4.circom

```diff
@@ -0,0 +1,58 @@
+pragma circom 2.1.8;
+
+include "circomlib/circuits/babyjub.circom";
+include "circomlib/circuits/poseidon.circom";
+include "gpc-util.circom";
+
+/**
+ * Module for establishing ownership of a POD object by the holder of
+ * a Semaphore V4 identity.  The owner is identified by their EdDSA
+ * public key, which should be externally constrained to an entry of
+ * an object.  Ownership is proven by presenting the secret scalar
+ * (derivable from the corresponding private key) as a private input.
+ * 
+ * Optionally calculates a nullifer tied to the owner's identity.  The
+ * revealed nullifier will be set to -1 if unused.
+ * 
+ * This module can be disabled entirely using the `enabled` signal, in
+ * which case all inputs are unconstrained.
+ *
+ * Boolean signals are assumed (but not constrained) to be 0 or 1.
+ * They will usually be implicitly constrained when they are unpacked
+ * using Num2Bits.
+ */
+template OwnerModuleSemaphoreV4 () {
+    // Global flag to enable or disable ownership constraints.
+    signal input enabled;
+    // Constrained externally: enabled * (1 - enabled) === 0;
+
+    // Secret scalar of owner's Semaphore V4 identity kept hidden and
+    // verified.
+    signal input secretScalar;
+
+    // Owner's identity commitment (public key hash) to be verified.
+    signal input identityCommitmentHash;
+
+    // Verify semaphore private secrets match the commitment in the POD by
+    // re-generating the public commitment to compare.
+    signal (computedPublicKeyX, computedPublicKeyY) <== BabyPbk()(secretScalar);
+    signal computedIdentityCommitment <== Poseidon(2)([computedPublicKeyX, computedPublicKeyY]);
+    signal computedIdentityCommitmentHash <== Poseidon(1)([computedIdentityCommitment]);
+
+    (computedIdentityCommitmentHash - identityCommitmentHash) * enabled === 0;
+
+    // External nullifier, used to tie together nullifiers within a single category. 
+    signal input externalNullifier;
+
+    // Calculate nullifier (privately).
+    signal nullifierHash <== Poseidon(2)([externalNullifier, secretScalar]);
```

Comment:
These inputs are in the same order as the Semaphore circuit, so it should be the same nullifier.
That's good, I'm just confirming that I double-checked it.